zulipterminal/model.py

```diff
@@ -853,6 +835,10 @@ def _update_topic_index(self, stream_id: int, topic_name: str) -> None:
         Update topic order in index based on incoming message.
         Helper method called by _handle_message_event
         """
+        # Fetch and index topics before updating the index if they have not
+        # been fetched yet via the topic toggle.
+        self.set_initial_topics_index(stream_id)
+
         topic_index = self.index['topics'][stream_id]
```

Comment:
Using a method in this iteration is a good first step, but note that in both cases the code is setting the initial index - ensuring it exists or is fetched - then looking it up for a particular stream. This is a standard pattern where one object handles caching/synchronizing internally, but the external users of that code don't need to consider how that happens, or even be aware of it - they just want the list of topics! Of course, there can be cases where the external code might want this (eg. observer pattern) but we don't need it here, at least yet.